### PR TITLE
Merge _build_* workflows into a single YAML file

### PR DESCRIPTION
**Pull request by olupton**
Introduce a generic `_build.yaml` that could replace `_build_t5x.yaml`, `_build_pax.yaml`, etc.

This will also be used in https://github.com/NVIDIA/JAX-Toolbox/pull/423, and helps keep us below GitHub's limitations on number of reusable workflows.

### CODE REVIEW

File: .github/workflows/_build.yaml [hunk 1]
```diff
@@ -19,21 +19,27 @@ on:
...
-        description: 'Name of the endpoint JSON file for shields.io badge'
-        required: false
-        default: 'badge-pax-build'
+        description: 'Name of the endpoint JSON file for shields.io badge, e.g. badge-t5x-build'
+        required: true
+      CONTAINER_NAME:
+        type: string
+        description: "Container name, e.g. upstream-t5x"
+        required: true
+      DOCKER_FILE:
```
Comment (yhtang) [inline .github/workflows/_build.yaml:32]: Please use `DOCKERFILE` with no underscore in between.

**Review by yhtang — DISMISSED** [review]
LGTM. Just one minor pick 😄

File: .github/workflows/_build.yaml (hunk 1 above)
Comment (olupton) [reply .github/workflows/_build.yaml]: Done :)

**Comment by terrykong** [comment]
Do you plan on removing `_build_pax.yaml` in this PR?

**Comment by olupton** [comment]
> Do you plan on removing `_build_pax.yaml` in this PR?

That's already done, GitHub is just choosing to express it as a rename from `_build_pax.yaml` to `_build.yaml`.

**Review by yhtang — APPROVED** [review]

**Pull request merged**